bindings/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,201 @@
+set(TARGET_NAME svs_runtime)
+
+set(SVS_RUNTIME_HEADERS
+    include/IndexSVSImplDefs.h
+    include/IndexSVSFlatImpl.h
+    include/IndexSVSVamanaImpl.h
+    include/training.h
+    include/vamana_index.h
+    include/dynamic_vamana_index.h
+)
+
+set(SVS_RUNTIME_SOURCES
+    src/IndexSVSImplUtils.h
+    src/svs_runtime_utils.h
+    src/dynamic_vamana_index_impl.h
+    src/IndexSVSFlatImpl.cpp
+    src/IndexSVSVamanaImpl.cpp
+    src/training.cpp
+    src/dynamic_vamana_index.cpp
+)
+
+option(SVS_RUNTIME_ENABLE_LVQ_LEANVEC "Enable compilation of SVS runtime with LVQ and LeanVec support" ON)
+if (SVS_RUNTIME_ENABLE_LVQ_LEANVEC)
+    message(STATUS "SVS runtime will be built with LVQ support")
+    list(APPEND SVS_RUNTIME_HEADERS
+        include/IndexSVSVamanaLVQImpl.h
+        include/IndexSVSVamanaLeanVecImpl.h
+    )
+    list(APPEND SVS_RUNTIME_SOURCES
+        src/IndexSVSVamanaLVQImpl.cpp
+        src/IndexSVSVamanaLeanVecImpl.cpp
+    )
+else()
+    message(STATUS "SVS runtime will be built without LVQ or LeanVec support")
+endif()
+
+add_library(${TARGET_NAME} SHARED
+    ${SVS_RUNTIME_HEADERS}
+    ${SVS_RUNTIME_SOURCES}
+)
+
+target_include_directories(${TARGET_NAME} PRIVATE
+    ${CMAKE_CURRENT_SOURCE_DIR}/include
+)
+
+find_package(OpenMP REQUIRED)
+target_link_libraries(${TARGET_NAME} PUBLIC OpenMP::OpenMP_CXX)
+
+target_compile_options(${TARGET_NAME} PRIVATE
+    -DSVS_ENABLE_OMP=1
+    -fvisibility=hidden
+)
+
+if(UNIX AND NOT APPLE)
+    # Don't export 3rd-party symbols from the lib
+    target_link_options(${TARGET_NAME} PRIVATE "SHELL:-Wl,--exclude-libs,ALL")
+endif()
+
+target_compile_features(${TARGET_NAME} INTERFACE cxx_std_20)
+set_target_properties(${TARGET_NAME} PROPERTIES PUBLIC_HEADER "${SVS_RUNTIME_HEADERS}")
+set_target_properties(${TARGET_NAME} PROPERTIES CXX_STANDARD 20)
+set_target_properties(${TARGET_NAME} PROPERTIES CXX_STANDARD_REQUIRED ON)
+set_target_properties(${TARGET_NAME} PROPERTIES CXX_EXTENSIONS OFF)
+set_target_properties(${TARGET_NAME} PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR} )
+
+if(DEFINED SVS_LVQ_HEADER AND DEFINED SVS_LEANVEC_HEADER)
+    # expected that pre-defined headers are implementation headers
+    message(STATUS "Using pre-defined LVQ header: ${SVS_LVQ_HEADER}")
+    message(STATUS "Using pre-defined LeanVec header: ${SVS_LEANVEC_HEADER}")
+else()
+    set(SVS_LVQ_HEADER "svs/extensions/vamana/lvq.h")
+    set(SVS_LEANVEC_HEADER "svs/extensions/vamana/leanvec.h")
+endif()
+
+if ((SVS_RUNTIME_ENABLE_LVQ_LEANVEC))
+    if(RUNTIME_BINDINGS_PRIVATE_SOURCE_BUILD)
+        message(STATUS "Building directly from private sources")
+        target_link_libraries(${TARGET_NAME} PRIVATE
+            svs::svs
+        )
+        link_mkl_static(${TARGET_NAME})
+    elseif(TARGET svs_static_library)
+        # Links to SVS static library built as part of the main SVS build
+        target_link_libraries(${TARGET_NAME} PRIVATE
+            svs_devel
```

Comment:
What is svs_devel? Is there a reason linking here would be different than lines 131-135?

Reply:
Originally, this branch was made to cover the case "when building from private sources" - in such case it waas expected that `svs_static_library` target is defined.
The change (40177d831ccb2bf0540723e1d25d7d8dc0b4bc6b, author: @ethanglaser) introduced dedicated variable for the case and made this branch useleass.
Seems like, this branch can be removed.